### ISSUE TITLE
When I train yolov8l-cls, Closing dataloader mosaic is not work.

### ISSUE THREAD

**Issue by Four-D** [ultralytics/ultralytics#16407]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

1.use following code in train.py file ,and use ultralytics==8.2.98.When epoch is 5, My Terminal is not output "Closing dataloader mosaic is not work."

model = YOLO(r"yolov8l-cls.pt")
model.train(data='D:\\datasets\\aaa\\',  epochs=8, imgsz=320,workers=0,batch=128,close_mosaic=3)

2.use ultralytics>=8.1.0 is not work.
3.When I use ultralytics==8.0.x is work well

### Environment

python 3.11 
torch 2.4.1

### Minimal Reproducible Example

model = YOLO(r"yolov8l-cls.pt")
model.train(data='D:\\datasets\\aaa\\',  epochs=8, imgsz=320,workers=0,batch=128,close_mosaic=3)

### Additional

<img width="859" alt="图片_20240921163536" src="https://github.com/user-attachments/assets/f0de7cbe-9a20-40d7-b17a-9806131e2009">

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Four-D, thank you for bringing this to our attention 🚀! This is an automated response, but an Ultralytics engineer will assist you soon.

We recommend first checking out our [Docs](https://docs.ultralytics.com) for guidance on using our library effectively, which may address common questions or issues. You can explore [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples for further insights.

If this is a 🐛 Bug Report, thank you for providing a [minimal reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This is extremely helpful in diagnosing the issue.

For optimal performance and compatibility, ensure you are using the latest version of the `ultralytics` package. You can upgrade easily with:

```bash
pip install -U ultralytics
```

For real-time conversations, join our [Discord community](https://ultralytics.com/discord) 🎧. You can also engage in our [Discourse forum](https://community.ultralytics.com) or on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade Environment

Ensure your setup aligns with these verified environments—complete with dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud:** [GCP Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon:** [AWS Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker:** [Docker Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status Check

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge indicates that all CI tests are passing successfully. CI verifies all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across various systems every 24 hours.

Thank you for contributing to Ultralytics and your willingness to submit a PR! Your input helps us improve continuously 🌟.

**Comment by Y-T-G** [comment]
Classification models don't use mosaic augmentation.
